textalyze.js

```diff
@@ -29,40 +31,116 @@ function itemCounts(array) {
  */
 function sanitize(string) {
-  if (validateString(string)) {
+  if (!isString(string)) {
     return '';
   }
 
-  return string.toLowerCase();
+  return string.toLowerCase().replace(/[^a-z]/gi, '');
+}
+
+/**
+ * Given an input Map and total count of elements,
+ * returns the object containing the percentage acording to total
+ * @param {Map} counts - The map containing the counts for each value
+ * @param {Number} total - The total count of elements
+ * @param {Boolean} sorted - If the return must be sorted
+ * @returns {Map} The frequency in percentage of each value
+ */
+function getFrequencyStatistics(counts, total, sorted) {
+  if (!counts || !(counts instanceof Map)) {
+    return counts;
+  }
+
+  if (!total || Number.isNaN(Number(total))) {
+    return counts;
+  }
+
+  let statistics = new Map();
+
+  counts.forEach((value, key) => {
+    const percentage = value / total;
+    statistics.set(key, percentage);
+  });
+
+  if (sorted) {
+    statistics = new Map([...statistics.entries()].sort());
+  }
+
+  return statistics;
 }
 
-module.exports = { itemCounts, arrayFrom, sanitize };
+/**
+ * Builds a string containing repetitions of histogramString
+ * according to current value in comparison to maxValue
+ * @param {Number} value - The current value
+ * @param {Number} maxValue - The max value for param value
+ * @param {String} histogramString - The string to compose the histogram bar
+ * @param {Number} maxBarLength - The max amount of times histogramString can be repeated
+ * @returns {String} The hitogram bar
+ */
+function getHistogramBar(value, maxValue, histogramString, maxBarLength) {
+  const charCount = value * maxBarLength / maxValue;
+  const histogramBar = histogramString.repeat(charCount);
+
+  return histogramBar;
+}
+
+module.exports = {
+  itemCounts,
+  arrayFrom,
+  sanitize,
+  getFrequencyStatistics,
+  getHistogramBar,
+};
 
 //
 // running the app
 //
-if (require.main === module) {
-  const string = 'Hello World';
+/**
+ * Runs the main application
+ * @param {String []} args - The args
+ */
+function main(argv) {
+  const path = argv[2];
+  const histogramString = argv[3] || '-';
+
+  fs.readFile(path, (err, data) => {
+    if (err) {
+      console.error(err);
+      return;
+    }
+
+    const sanitizedString = sanitize(data.toString());
+    const array = arrayFrom(sanitizedString);
+    const counts = itemCounts(array);
+    const sorted = true;
+    const statistics = getFrequencyStatistics(counts, array.length, sorted);
+    const maxValue = Math.max(...statistics.values());
 
-  console.log(`The counts for "${string}" are...`);
+    console.log(`The counts for "${path}" are...`);
 
-  const sanitizedStirng = sanitize(string);
-  const array = arrayFrom(sanitizedStirng);
+    statistics.forEach((value, key) => {
+      const humanReadablePercentage = (value * 100).toFixed(2).padStart(5, ' ');
+      const histogramBar = getHistogramBar(value, maxValue, histogramString, 80);
```

Comment:
Check out [process.stdout.columns](https://nodejs.org/api/tty.html#tty_writestream_columns), which will return the width of the current window in columns or `undefined` if we're not printing to the terminal (e.g., piping the output to another process).